compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -30,29 +30,47 @@ static LogicalResult setRootConfigForPadPipeline(func::FuncOp entryPointFn,
       IREE::Codegen::DispatchLoweringPassPipeline::None);
 }
 
-static SmallVector<int64_t> getPackedSize(Operation *op, const int packLevel) {
+/// Returns the largest number that perfectly divides `num` that is less than or
+/// equal to max
+static int findLargestFactor(int num, int max) {
+  if (num < max) {
+    return num;
+  }
+  for (int i = max; i > 0; i--) {
+    if (num % i == 0) {
+      return i;
+    }
+  }
+  return 1;
+}
+
+static SmallVector<int64_t> getPackedSize(linalg::LinalgOp linalgOp,
+                                          const int packLevel, int m = 0,
+                                          int n = 0, int k = 0) {
   // TODO: consider emiting an error/warning if the default sizes are used as a
   // fallback.
   SmallVector<int64_t> defaultSizes;
```

Comment:
If the '4,4,8' is related to vector sizes for matmul inner loop, the optimal sizes (element type dependent) can be obtained from getAIEMatmulInstructionSize

(not a request for change here as the 4,4,8, are here already, just some info).

Reply:
Currently getAIEMatmulInstructionSize this is not returning a config for i32 case, perhaps because there is no vector intrinsic for it, so we need to pick some packing size for such cases separately I think.